pkg/analysis/passes/checksum/checksum_test.go

```diff
@@ -0,0 +1,277 @@
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 "))
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestValidMD5ChecksumFromUrlWithSpaces(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "f9b1c42c45cbf4953d7da5c31b3d73d9 archive-file-name.zip"))
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+func TestInvalidMD5ChecksumFromUrl(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(200, "x9b1c42c45cbf4953d7da5c31b3d73d3"))
+
+	_, err := Analyzer.Run(pass)
+	assert.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 1)
+	assert.Equal(
+		t,
+		"The provided checksum x9b1c42c45cbf4953d7da5c31b3d73d3 does not match the plugin archive",
+		interceptor.Diagnostics[0].Title,
+	)
+}
+
+func TestWrongUrl404(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+
+	pass := &analysis.Pass{
+		ResultOf: map[*analysis.Analyzer]interface{}{},
+		CheckParams: &analysis.CheckParams{
+			Checksum:              "https://example.com/checksum",
+			ArchiveCalculatedMD5:  "f9b1c42c45cbf4953d7da5c31b3d73d9",
+			ArchiveCalculatedSHA1: "187cd80948f240957bf8399745335b89f005c5f0",
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "https://example.com/checksum",
+		httpmock.NewStringResponder(404, ""))
```

Comment:
```suggestion
	httpmock.RegisterResponder("GET", "https://example.com/checksum",
	httpmock.NewStringResponder(404, ""))
```

Reply:
Likewise. this is not an indentation error but autoformatted by https://github.com/segmentio/golines